test/integration/proc/redis/main_test.go

```diff
@@ -545,14 +574,37 @@ func TestMain(m *testing.M) {
 		addr, _ := net.ResolveTCPAddr("tcp4", fmt.Sprintf("127.0.0.1:%d", port))
 		nodes = append(nodes, addr)
 	}
-	if err := defaultClusterManager.Start(nodes...); err != nil {
-		logger.Fatal(err)
+	testWithNewCluster := func(fn func()) {
+		if err := defaultClusterManager.Start(nodes...); err != nil {
+			logger.Fatal(err)
+		}
+		defer func() {
+			if err := defaultClusterManager.Shutdown(); err != nil {
+				log.Fatalf("failed to shutdown cluster manager, err: %v", err)
+			}
+		}()
+		initRedisClient()
+		defer func() {
+			if err := c.Close(); err != nil {
+				log.Fatalf("failed to close redis client, err: %v", err)
+			}
+		}()
+		fn()
 	}
+
 	setupRedisProxy()
-	initRedisClient()
 
-	code := m.Run()
+	var code = 0
+	defer os.Exit(code)
+	testWithNewCluster(func() {
+		code = m.Run()
+	})
+	if code != 0 {
+		return
+	}
 
-	defaultClusterManager.Shutdown()
-	os.Exit(code)
+	setCompress(true, 1)
+	testWithNewCluster(func() {
+		code = m.Run()
+	})
```

Comment:
If use table-driven tests to combine these, it will be more easy to add another in the future.